### ISSUE TITLE
Investigate support for Canvas, Blackboard, etc

### ISSUE THREAD

**Issue by choldgraf** [2i2c-org/infrastructure#1676]
### Context

Currently, our infrastructure can connect with _github repositories_ to pull in content via nbgitpuller. However, many (most?) organizations do not feel comfortable using GitHub for their course content. Instead they use a course management system like Canvas or Blackboard.

There have already been a few cases where others have asked us to connect a hub with Canvas (e.g. I believe Toronto wanted this, and Berkeley uses Canvas as well).

### Proposal

We should investigate whether it is feasible to "officially support" certain kinds of course management systems. If so, we should document their use so that others know to ask us to build these integrations.

A few questions to consider:

- Are there any hard technical issues for integrating with these?
- What is the subset of course management systems that is used by the majority of the communities we'd want to serve?
- How hard would it be to maintain these integrations over time?
- Would new development be needed to make this work? If so can we find communities that are willing to pay for this work?

**Comment by jmunroe** [comment]
I was asked today by a UofT faculty member (as part of UofT community event) whether 2i2c is looking at "officially" support Canvas and integrate it with our Hub. I replied that it is on our product roadmap with no specific timeline.  

In order to move this just a little bit forward, I checked out Infrastructure's (the company that that creates Canvas), Integration Partner program just to see how involved the process would be and what types of support are available.  

link: https://www.instructure.com/partners/our-partners

I dropped in my contact info into form on behalf of 2i2c with the expectation some from their team will reach out over email.  

This is only information gathering at this point: what is their "Integration Partner" program, what are the terms and conditions, what are the costs, what are the benefits, etc.

One feature I note is the ability to subscribe to "Canvas Sandbox - **$2,000/Year**"

> Canvas Sandbox: A hosted account to access full Canvas functionality for demonstrations, developing, and testing. Includes a beta instance where you can test your integration before new features are deployed in Canvas production instances. Purchase of a Canvas Sandbox includes access to partner support.
